### PR TITLE
Add fine tune example for bge model

### PR DESCRIPTION
**Pull request by lu-wang-dl**
Manually test on https://adb-7064161269814046.2.staging.azuredatabricks.net/?o=7064161269814046#notebook/892486914846105/command/892486914846109

### CODE REVIEW

File: llm-models/embedding/bge/bge-large/04_fine_tune_embedding.py.py
Comment (es94129) [inline llm-models/embedding/bge/bge-large/04_fine_tune_embedding.py.py:1]: nit: The filenames of the 03 and 04 notebooks have 2 `.py` at the end

File: llm-models/embedding/bge/bge-large/04_fine_tune_embedding.py.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+from sentence_transformers import SentenceTransformer
+model = SentenceTransformer('BAAI/bge-large-en')
+
+# COMMAND ----------
+
+from torch.utils.data import DataLoader
+from sentence_transformers import losses
+from accelerate import notebook_launcher
+
+train_dataloader = DataLoader(train_examples, shuffle=True, batch_size=3)
```
Comment (es94129) [inline llm-models/embedding/bge/bge-large/04_fine_tune_embedding.py.py:89]: Curious about how the batch size is selected?

File: llm-models/embedding/bge/bge-large/04_fine_tune_embedding.py.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+import mlflow
+
+with mlflow.start_run() as run:
+  my_model = SentenceTransformerEmbeddingModel()
+  model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=my_model, input_example=["London is known for its finacial district"], artifacts={"sentence-transformer-model": output_model_path}, conda_env=EMBEDDING_CONDA_ENV)
+
+# COMMAND ----------
+
+# Register model
+# This may take 2.2 minutes to complete
```
Comment (es94129) [inline llm-models/embedding/bge/bge-large/04_fine_tune_embedding.py.py:146]: nit: 1 minute

**Review by es94129 — APPROVED** [review]
LGTM! Quite interesting for me to learn how embeddings are fine-tuned.

File: llm-models/embedding/bge/bge-large/04_fine_tune_embedding.py.py (hunk 1 above)
Comment (rithwik-db) [reply llm-models/embedding/bge/bge-large/04_fine_tune_embedding.py.py:89]: I guess it's just the `batch_size` that doesn't OOM...

File: llm-models/embedding/bge/bge-large/04_fine_tune_embedding.py.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    ]
+)
+
+# COMMAND ----------
+
+import mlflow
+
+with mlflow.start_run() as run:
+  my_model = SentenceTransformerEmbeddingModel()
+  model_info = mlflow.pyfunc.log_model(artifact_path="model", python_model=my_model, input_example=["London is known for its finacial district"], artifacts={"sentence-transformer-model": output_model_path}, conda_env=EMBEDDING_CONDA_ENV)
```
Comment (rithwik-db) [inline llm-models/embedding/bge/bge-large/04_fine_tune_embedding.py.py:141]: Do we want to add an additional query example with the prefix text "Represent this sentence for searching relevant passages: "?

**Pull request merged**